src/Core/Config/ValueObject/Cache.php

```diff
@@ -223,4 +279,71 @@ public function keyDiff(array $config): array
 
 		return $return;
 	}
+
+	/**
+	 * Merges a new Cache into the existing one and returns the merged Cache
+	 *
+	 * @param Cache $cache The cache, which should get merged into this Cache
+	 *
+	 * @return Cache The merged Cache
+	 */
+	public function merge(Cache $cache): Cache
+	{
+		$newConfig = $this->config;
+		$newSource = $this->source;
+
+		$categories = array_keys($cache->config);
+
+		foreach ($categories as $category) {
```

Comment:
This prevents adding new categories to the current cache from the merged cache.

Reply:
No it doesn't .. I now added some unit tests :)
- I take the whole config array from `$this->config`
- I loop over all categories of `$cache->config`

There's no gap as far as I can see.

Reply:
Sorry, you're right, I misread.